### DISCUSSION TITLE
Searching date ranges with common phrases

### DISCUSSION THREAD

**Discussion by jordan-mcrae** [algolia/instantsearch#5343] [Q&A] [answered]
Hi! I'm new to Algolia and am just trying to evaluate if it's something that might be useful for my project.

Does anyone have an example or best practice for setting up date queries based on common phrases such as "today", "tomorrow", "next week", etc?

For example, if I had indexed a list of sports games and I wanted my users to be able to search for a specific game such as:

"Leafs vs. Canadiens next week", where "Leafs" and "Canadiens" would both be a "team", and "next week" would dynamically search for a date field between next Monday + Sunday.

Thanks for the help!

**Comment by Haroenv** [comment, marked as answer]
If you store the date in the record, like `20221103` (important that this is an ascending number), you can use it in filters. From the frontend this is simple and can be using `connectRange`.

If you want to automatically respond to queries that say "next week", you can add a new query rule daily that looks like this:

```json
{
  "objectID": "next-week-filter",
  "conditions": [{
    "anchoring": "contains",
    "pattern": "next week"
  }],
  "consequence": {
    "params": {
      "filters": "date:20221103 TO 20221110"
    }
  }
}
```

**Reply by jordan-mcrae** [reply]
Thank you! This is super helpful.
